wordpress-mobile/WordPress-Android
# Comment refactor Part 1

Currently, there are two entry points to the screen of Comments Details:
1. Notifications List -> Comment
2. My Site -> Comments List -> Comment

For some reason, we implemented every thing in that big `CommentDetailFragment`. There is some logic that is only triggered by specific entry points. To maintain the simplicity of the parent class, I suggest relocating this code to a subclass. Furthermore, converting subclasses to Kotlin can enhance development efficiency.

This a PR contains:

1. Create two Fragments (which extend `CommentDetailFragment`) for different purpose.
2. Convert a file to Kotlin.
3. Remove some redundant logic

There will be follow-up PRs for furthering refactors.

-----

## To Test:

1. Sign in JP app
2. Go to the Notifications tab
3. Click on `Comment` type of notification
4. Every thing should be the same as on the `feature/notifications_refresh_p2` branch.
5. Go to My Site -> Comments, click on a comment
7. Every thing should be the same as on the `feature/notifications_refresh_p2` branch.
8. Done, thank you.

-----

## Regression Notes

1. Potential unintended areas of impact

    - notifications, comment

11. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

12. What automated tests I added (or what prevented me from doing so)

    - none

-----

## PR Submission Checklist:

Skipped

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

Skipped
